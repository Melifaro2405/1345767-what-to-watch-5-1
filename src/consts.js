const ALL_GENRES = `All genres`;
const COUNT_SHOWN_GENRES = 10;
const COUNT_SHOWN_FILMS = 8;
const COUNT_LIKE_GENRE_FILMS = 4;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const CommentLength = {
  MIN: 50,
  MAX: 400
};

const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  FILM_BY_ID: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  PLAYER: `/player`,
  PLAYER_BY_ID: `/player/:id`,
};

const APIRoute = {
  FILMS: `/films`,
  FAVORITE_FILMS: `/favorite`,
  PROMO_FILM: `/films/promo`,
  COMMENTS: `/comments`,
  LOGIN: `/login`
};

export {
  ALL_GENRES,
  COUNT_SHOWN_GENRES,
  COUNT_SHOWN_FILMS,
  COUNT_LIKE_GENRE_FILMS,
  AuthorizationStatus,
  CommentLength,
  AppRoute,
  APIRoute
};
