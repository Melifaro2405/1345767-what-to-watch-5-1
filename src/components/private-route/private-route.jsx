import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute, AuthorizationStatus} from "../../consts";


const PrivateRoute = ({render, path, authorizationStatus}) => {
  return (
    <Route exact path={path} render={(routeProps) => {
      return (
        authorizationStatus === AuthorizationStatus.AUTH ? render(routeProps) : <Redirect to={AppRoute.LOGIN} />
      );
    }}/>
  );
};

PrivateRoute.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
