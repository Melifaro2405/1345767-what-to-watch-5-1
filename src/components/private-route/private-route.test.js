import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {PrivateRoute} from "./private-route";
import {AppRoute, AuthorizationStatus} from "../../consts";

const MockComponent = () => <div />;

describe(`Should PrivateRoute render correctly`, () => {
  test.each([
    [`with`, AuthorizationStatus.AUTH],
    [`without`, AuthorizationStatus.NO_AUTH],
  ])(`%s authorization`, (_expected, authorizationStatus) => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <PrivateRoute
              path={AppRoute.ROOT}
              authorizationStatus={authorizationStatus}
              render={MockComponent}
            />
          </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
