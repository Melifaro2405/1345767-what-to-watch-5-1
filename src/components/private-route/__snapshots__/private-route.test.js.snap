// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PrivateRoute render correctly with authorization 1`] = `<div />`;

exports[`Should PrivateRoute render correctly without authorization 1`] = `null`;
